fix(move): prevent duplicate copying of tracks and extras when processing albums

ali-ramadhan opened the pull request:
- [x] I have read the contributing guide in the documentation.

### Description

Updated the `edit_new_items` function to skip copying tracks and extras if their album is already being processed, ensuring no duplicates occur.

This PR is addressing the issue raised in discussion #280.

<!-- readthedocs-preview mrmoe start -->
----
📚 Documentation preview 📚: https://mrmoe--294.org.readthedocs.build/en/294/

<!-- readthedocs-preview mrmoe end -->

jtpavlock commented on moe/move/move_core.py:
```diff
@@ -68,7 +68,15 @@ def add_config_validator(settings: dynaconf.base.LazySettings):
+        # Skip tracks and extras if their album is also being processed.
+        # This prevents double-copying since _copy_album already handles all
+        # tracks/extras.
+        if isinstance(item, (Track, Extra)) and item.album._id in album_ids_in_items:
```
You should just be able to do something like

```python
        if not (isinstance(item, (Track, Extra)) and item.album in items):
            copy_item(item)
```

And then you don't need to deal with using the ids or the continue statement. If you had the logic reversed with the continue statement for clarity that's fine too though

ali-ramadhan replied:
Actually I do think this is clearer! Will make the change.

jtpavlock commented on tests/move/test_move_core.py:
```diff
@@ -501,6 +501,20 @@ def test_extra(self, mock_copy):
+        album = album_factory(num_tracks=2, num_extras=2)
+        mock_session = MagicMock()
+
+        items = [album] + list(album.tracks) + list(album.extras)
```
tracks and extras are already lists so you don't need list() here

ali-ramadhan replied:
Fixed!

ali-ramadhan commented on tests/move/test_move_core.py:
```diff
@@ -501,6 +501,20 @@ def test_extra(self, mock_copy):
+    def test_album_with_tracks_and_extras_no_duplicate_copy(self, mock_copy):
+        """When album and its tracks/extras are in same batch, avoid duplicate copying.
+
+        See: https://github.com/orgs/MoeMusic/discussions/280
```
Will remove the mention in the docstring and move it to the commit message.

I think the reason I do this is to provide more context on why the test was written. And I like being able to quickly read up on the original motivation for the test. It might get lost in git history after renaming/refactoring stuff so I tend to keep it in comments. But I understand this could lead to lots of links littering the test suite!

jtpavlock replied:
Yeah, that's fair too. I guess I'd at least like it in the commit message, but I don't mind as much if it's also added in the test descriptions.

ali-ramadhan commented:
> Sorry for the nitpicks, thanks for the pr!

Not at all, thank you for the thorough review! I've addressed your comments so just requesting another review.
